react/src/js/react/posts/Post.jsx

```diff
@@ -1,53 +1,87 @@
 import React from "react";
+import Reflux from "reflux";
 import PropTypes from "prop-types";
 
-import { Thumbnail } from "react-bootstrap";
+import { PostsStore, PostsActions } from "./PostsStore";
+import { Thumbnail, Button } from "react-bootstrap";
 
 import { formatDate } from "../util/DateUtil";
 
 /**
  * Displays a singular post
  */
-export default class Post extends React.Component {
+export default class Post extends Reflux.Component {
     static propTypes = {
-        post: PropTypes.object.isrequired
+        className: PropTypes.string.isRequired,
+        post: PropTypes.object.isRequired
+    }
+
+    constructor(props) {
+        super(props);
+        this.store = PostsStore;
+    }
+
+    // TODO Need to render properly
+    // From VinayC, https://stackoverflow.com/questions/8499633/how-to-display-base64-images-in-html
+    renderContent = () => {
+        const { contentType, content } = this.props.post;
+        if (contentType === ("image/png;base64" || "image/jpeg;base64")) {
+            const name = `data:${contentType},${content}`;
+            return <img src={name} />;
+        } else {
+            return <p className="content">{content}</p>;
+        }
     }
 
     // TODO
     renderComments() {
         return null;
+    }
+
+    renderFooter() {
+        const { post } = this.props;
+        return (
+            <div className="post-footer">
+                <p className="categories">{post.categories}</p>
+                { post.source ? <a href={post.source} className="source">source</a> : null}
+                { post.origin ? <a href={post.origin}>origin</a> : null }
+            </div>
+        );
+    }
 
-        /* <div className="comment-wrapper">
-			{this.state.comments.map(comment => (
-				<p className="comment" key={comment}>comment</p>
-			))}
-			<div className="comment-feild">
-				<textarea className="comment-input" id={this.props.postId} name="text" rows="1" wrap="soft" />
-				<button className="send-comment-button" onClick={this.makeComment}>
-					<i className="fa fa-paper-plane send-comment-button" />
-				</button>
-			</div>
-		</div> */
+    handleDeletePost = () => {
+        PostsActions.deletePost(this.props.post.id, this.props.post.post_id);
     }
 
     render() {
+        const { post } = this.props;
         return (
             <Thumbnail>
+                <div className="post-header">
+                    <i className="fas fa-user-lock">{post.visibility}</i>
+                    <Button
```

Comment:
Shouldn't this only be visible if the post is made by the currently signed in user?

Reply:
I'm just leaving it for testing